lib/src/utils/parsing/parsing_utils.dart

```diff
@@ -0,0 +1,321 @@
+    ? null
+    : parseUint64Str(tx.validityIntervalStart, {}, InvalidDataReason.validityIntervalStartInvalid);
+
+  // mint instructions
+  final mint = tx.mint == null ? null : parseTokenBundle(tx.mint, false, parseInt64_str);
+
+  // script data hash hex
+  final scriptDataHashHex = tx.scriptDataHash == null 
+    ? null
+    : parseHexStringOfLength(tx.scriptDataHash, scriptDataHashLength, InvalidDataReason.scriptDataHashWrongLength);
+
+  // collateral inputs
+  validate(isArray(tx.collateralInputs ?? []), InvalidDataReason.collateralInputsNotArray.message);
+  final collateralInputs = (tx.collateralInputs ?? []).map((inp) => parseTxInput(inp)).toList();
+
+  // required signers
+  validate(isArray(tx.requiredSigners ?? []), InvalidDataReason.requiredSignersNotArray.message);
+  final requiredSigners = (tx.requiredSigners ?? []).map((rs) => parseRequiredSigner(rs)).toList();
+
+  // include network ID
+  final includeNetworkId = tx.includeNetworkId == null 
+    ? false
+    : parseBoolean(tx.includeNetworkId, InvalidDataReason.networkIdIncludeInvalid.message);
+
+  // collateral output
+  final collateralOutput = tx.collateralOutput == null
+    ? null
+    : parseTxOutput(tx.collateralOutput, tx.network);
+  validate(collateralOutput?.datum == null, InvalidDataReason.collateralInputContainsDatum.message);
+  validate(collateralOutput?.referenceScriptHex == null, InvalidDataReason.collateralInputContainsReferenceScript.message);
+
+  // total collateral  
+  final totalCollateral = tx.totalCollateral == null
+    ? null  
+    : parseCoin(tx.totalCollateral, InvalidDataReason.totalCollateralNotValid);
+
+  // reference inputs
+  validate(isArray(tx.referenceInputs ?? []), InvalidDataReason.referenceInputsNotArray.message);
+  final referenceInputs = (tx.referenceInputs ?? []).map((ri) => parseTxInput(ri)).toList();
+
+  // voting procedures
+  validate(isArray(tx.votingProcedures ?? []), InvalidDataReason.votingProceduresNotArray.message);
+  final votingProcedures = (tx.votingProcedures ?? []).map((x) => parseVoterVotes(x)).toList();
+  validate(votingProcedures.length <= 1, InvalidDataReason.votingProceduresInvalidNumberOfVoters.message);
+  for (final voterVotes in votingProcedures) {
+    validate(voterVotes.votes.length == 1, InvalidDataReason.votingProceduresInvalidNumberOfVotes.message);
+  }
+
+  // treasury
+  final treasury = tx.treasury == null
+    ? null
+    : parseCoin(tx.treasury, InvalidDataReason.treasuryNotValid);
+
+  // donation
+  final donation = tx.donation == null
+    ? null
+    : parseUint64Str(tx.donation, {'min': '1', 'max': maxLovelaceSupplyStr}, InvalidDataReason.donationNotValid);
+
+  return ParsedTransaction(
+    network: network,
+    inputs: inputs,
+    outputs: outputs,
+    ttl: ttl == null ? null : BigInt.parse(ttl),
+    auxiliaryData: auxiliaryData,
+    validityIntervalStart: validityIntervalStart == null ? null : BigInt.parse(validityIntervalStart),
```

Comment:
`validityIntervalStart` should be BigInt in `Transaction` too. And you won't need to parse it here as BigInt.